### PR TITLE
fix set within Map child element

### PR DESCRIPTION
**Pull request by geseq**
#### Summary

closes https://github.com/stretchr/objx/issues/97

Fixes Set call when trying to set a value into an inner element that's a `Map` type as described in https://github.com/stretchr/objx/issues/97

#### Checklist
- [x] Tests are passing: `task test`
- [x] Code style is correct: `task lint`

### CODE REVIEW

**Comment by geseq** [comment]
@hanzei  another one

**Comment by geseq** [comment]
depends on some changes in #104 I'll update it again once that is merged

File: accessors.go [hunk 1]
```diff
@@ -114,6 +114,7 @@ func getKey(s string) (string, string) {
 // access accesses the object using the selector and performs the
 // appropriate action.
 func access(current interface{}, selector string, value interface{}, isSet bool) interface{} {
+
```
Comment (hanzei) [inline accessors.go:117]: Unrelated change?

Comment (geseq) [inline accessors.go:117]:
```suggestion
```

File: accessors.go [hunk 2]
```diff
@@ -130,17 +131,23 @@ func access(current interface{}, selector string, value interface{}, isSet bool)
 	if curMap, ok := current.(Map); ok {
 		current = map[string]interface{}(curMap)
 	}
+
```
Comment (geseq) [inline accessors.go:134]:
```suggestion
```

File: accessors.go [hunk 3]
```diff
@@ -130,17 +131,23 @@ func access(current interface{}, selector string, value interface{}, isSet bool)
...
 	// get the object in question
 	switch current.(type) {
 	case map[string]interface{}:
 		curMSI := current.(map[string]interface{})
+
```
Comment (geseq) [inline accessors.go:139]:
```suggestion
```

File: accessors.go [hunk 4]
```diff
@@ -130,17 +131,23 @@ func access(current interface{}, selector string, value interface{}, isSet bool)
...
 			return nil
 		}
 
 		_, ok := curMSI[thisSel].(map[string]interface{})
-		if (curMSI[thisSel] == nil || !ok) && len(indexes) == 0 && isSet {
+		if !ok {
+			_, ok = curMSI[thisSel].(Map)
+		}
+
+    if (curMSI[thisSel] == nil || !ok) && len(indexes) == 0 && isSet {
```
Comment (geseq) [inline accessors.go:150]:
```suggestion
		if (curMSI[thisSel] == nil || !ok) && len(indexes) == 0 && isSet {
```

**Review by geseq — COMMENTED** [review]
trying to do this on my phone

File: accessors.go (hunk 1 above)
Comment (geseq) [reply accessors.go]: yeah, looks like it

**Comment by geseq** [comment]
ready to review @hanzei

**Review by hanzei — APPROVED** [review]
